nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/WorkComputation.java

```diff
@@ -122,15 +122,24 @@ public void processRecord(ComputationContext context, String inputStreamName, Re
                         work.getId(), work.getTitle(), record));
                 context.askForCheckpoint();
             }
-            // Cleanup should take care of logging error, but better to dup this in debug
+            // Cleanup should take care of logging error except if exception comes from the cleanup
             log.debug("Exception during work " + work.getId(), e);
-            work.cleanUp(false, e);
+            // Try to cleanup after an exception, if exception comes from the previous cleanup it is a second try
```

Comment:
You mean that an exception at cleanup can trigger a second computation try?

Reply:
No, it means that the cleanup method might be called twice in the case the exception has been raised during the first cleanup done in AbstartWork#run. Should I reformulate?

Reply:
OK, yes please, clearer this way.